core/src/builtin/value.rs

```diff
@@ -0,0 +1,71 @@
+use crate::AllocId;
+use crate::ContextRef;
+use crate::OpRef;
+use crate::Type;
+
+pub struct Value<T: Into<Type> + From<Type>> {
+    op_id: AllocId,
+    ty: T,
+}
+
+impl<T: Into<Type> + From<Type> + Copy> Value<T> {
+    fn get_context(&self) -> ContextRef {
+        Type::try_from(self.get_type())
```

Comment:
Hmm... We have trait Ty, which should have get_context. Maybe we just require `T: Ty` then? And put all other magic inside Ty?

Reply:
Wair, maybe it's not a good idea. If we do that, we'll probably ban `Value<Type>`, which should be a thing, imo.

Reply:
Fine with me